sklearn/utils/_metadata_requests.py

```diff
@@ -368,13 +378,46 @@ class MetadataRequest:
     def __init__(self, owner):
         # this is used to check if the user has set any request values
         self._is_default_request = False
-        for method in METHODS:
+        self.owner = owner
+        for method in SIMPLE_METHODS:
             setattr(
                 self,
                 method,
                 MethodMetadataRequest(router=self, owner=owner, method=method),
             )
 
+    def __getattr__(self, name):
+        # Called when the default attribute access fails with an AttributeError
+        # (either __getattribute__() raises an AttributeError because name is
+        # not an instance attribute or an attribute in the class tree for self;
+        # or __get__() of a name property raises AttributeError). This method
+        # should either return the (computed) attribute value or raise an
+        # AttributeError exception.
```

Comment:
I am wondering if we could illustrate with small code snippets

Reply:
added a link to the official docs, This text is a copy paste from the official docs anyway.